### PR TITLE
Add CBMC proof for TaskPrioritySet

### PR DESCRIPTION
**Pull request by Baha**
Description
-----------
This PR adds the CBMC memory-safety proof for TaskPrioritySet.

Depends on #774

Checklist:
----------

- [ ] I have tested my changes. No regression in existing tests.
- [ ] My code is Linted.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: tools/cbmc/proofs/TaskPool/TaskPrioritySet/tasks_test_access_functions.h [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+ * This initialization is not guaranteed to be minimal, but it
+ * is quite efficient and it serves the same purpose
+ */
+BaseType_t xPrepareTaskLists( TaskHandle_t * xTask )
+{
+	TCB_t * pxTCB = NULL;
+
+	__CPROVER_assert_zero_allocation();
+
+	prvInitialiseTaskLists();
```
Comment (danielsn) [inline tools/cbmc/proofs/TaskPool/TaskPrioritySet/tasks_test_access_functions.h]: Using the init function can overconstrain the object.

File: tools/cbmc/proofs/TaskPool/TaskPrioritySet/tasks_test_access_functions.h [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+	pxTCB = xUnconstrainedTCB();
+
+	/* Nondet. insertion of another task in the same list to allow both cases in line 1656 (tasks.c) */
+	if ( nondet_bool() )
+	{
+		TCB_t * pxOtherTCB;
```
Comment (danielsn) [inline tools/cbmc/proofs/TaskPool/TaskPrioritySet/tasks_test_access_functions.h]: declare and init in one statement

File: tools/cbmc/proofs/TaskPool/TaskPrioritySet/tasks_test_access_functions.h [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+		pxOtherTCB = xUnconstrainedTCB();
+		if ( pxOtherTCB != NULL )
+		{
+			vListInsert( &pxReadyTasksLists[ pxOtherTCB->uxPriority ], &( pxOtherTCB->xStateListItem ) );
+		}
+	}
+
+	if ( pxTCB != NULL )
```
Comment (danielsn) [inline tools/cbmc/proofs/TaskPool/TaskPrioritySet/tasks_test_access_functions.h]: Can declare and init ```pcTCB``` here.

File: tools/cbmc/proofs/TaskPool/TaskPrioritySet/tasks_test_access_functions.h [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+	{
+		/* 
+		 * The `return pdFAIL` statement that follows can be uncommented
+		 * to test coverage when the task handle passed to TaskPrioritySet is NULL
+		 */
+		// return pdFAIL;
```
Comment (danielsn) [inline tools/cbmc/proofs/TaskPool/TaskPrioritySet/tasks_test_access_functions.h]: two comments in two styles?

File: tools/cbmc/proofs/TaskPool/TaskPrioritySet/tasks_test_access_functions.h [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+	{
+		return pdFAIL;
+	}
+
+	/* Nondet. insertion of this task in ready tasks list to allow both cases in line 1651 (tasks.c) */
+	if ( nondet_bool() )
+	{
+		vListInsert( &pxReadyTasksLists[ pxCurrentTCB->uxPriority ], &( pxCurrentTCB->xStateListItem ) );
+
+		/* Use of this macro ensures coverage on line 185 (list.c) */
```
Comment (danielsn) [inline tools/cbmc/proofs/TaskPool/TaskPrioritySet/tasks_test_access_functions.h]: This seems sketchy to me.

**Comment by markrtuttle** [comment]
Closed in favor of https://github.com/aws/amazon-freertos/pull/1325.

**Pull request closed without merging**